components/config-mgmt-service/integration_test/node_export_test.go

```diff
@@ -0,0 +1,498 @@
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+			},
+			allowedProjects:   []string{"project3", "project10", "project12", "project13"},
+			expectedNodeNames: []string{"1"},
+		},
+		{
+			description: "Two nodes with several projects where one matches one of several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "2",
+					},
+					Projects: []string{"project13", "project14", "project17", "project16"},
+				},
+			},
+			allowedProjects:   []string{"project3", "project10", "project12", "project13"},
+			expectedNodeNames: []string{"1", "2"},
+		},
+		{
+			description: "One node with several projects where none matches several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+			},
+			allowedProjects:   []string{"project14", "project10", "project12", "project13"},
+			expectedNodeNames: []string{},
+		},
+		{
+			description: "One node with several projects where two matches two of several requested project allowed",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName: "1",
+					},
+					Projects: []string{"project3", "project10", "project7", "project6"},
+				},
+			},
+			allowedProjects:   []string{"project3", "project10", "project12", "project13"},
+			expectedNodeNames: []string{"1"},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(fmt.Sprintf("Project filter: %s", test.description), func(t *testing.T) {
+			// Adding required node data
+			for index := range test.nodes {
+				test.nodes[index].Exists = true
+				test.nodes[index].NodeInfo.EntityUuid = newUUID()
+			}
+
+			// Add nodes with projects
+			suite.IngestNodes(test.nodes)
+			defer suite.DeleteAllDocuments()
+			ctx := auth_context.NewOutgoingContext(contextWithProjects(test.allowedProjects))
+			response, err := client.NodeExport(ctx, &request.NodeExport{
+				OutputType: "json",
+				Filter:     test.filter,
+			})
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			data := make([]byte, 0)
+			for {
+				tdata, err := response.Recv()
+				if err != nil && err == io.EOF {
+					data = append(data, tdata.GetContent()...)
+					break
+				}
+
+				require.NoError(t, err)
+				data = append(data, tdata.GetContent()...)
+			}
+
+			actualNodeNames := make([]string, 0)
+			jsonparser.ArrayEach(data, func(node []byte, _ jsonparser.ValueType, _ int, err error) {
+				if err == nil {
+					nodeName, err := jsonparser.GetString(node, "name")
+					if err == nil {
+						actualNodeNames = append(actualNodeNames, nodeName)
+					}
+				}
+			})
```

Comment:
I'm not sure I understand this part. Do we not have a type for the response returned here?

Reply:
We do not have an exposed type. The type is here https://github.com/chef/automate/blob/master/components/config-mgmt-service/grpcserver/node_export.go#L31. 

I found it easier to just parse the JSON.